ArduPlane/ArduPlane.cpp

```diff
@@ -845,43 +845,117 @@ void Plane::update_flight_stage(void)
   Do we think we are flying?
   Probabilistic method where a bool is low-passed and considered a probability.
 */
-void Plane::determine_is_flying(void)
+void Plane::update_is_flying_5Hz(void)
 {
     float aspeed;
-    bool isFlyingBool;
-
-    bool airspeedMovement = ahrs.airspeed_estimate(&aspeed) && (aspeed >= 5);
+    bool is_flying_bool;
+    uint32_t now_ms = hal.scheduler->millis();
+
+    float ground_speed_thresh_cm = (g.min_gndspeed_cm > 0) ? labs(g.min_gndspeed_cm*0.9f) : 500;
```

Comment:
This reads integer values and compares against integers, so if it's an uint32_t or some such you can avoid the labs and work in pure integer math.